day4/cards.go

```diff
@@ -0,0 +1,51 @@
+package day4
+
+import (
+	"bufio"
+	"math"
+	"os"
+	"strings"
+)
+
+// CountMatches counts the total number of matches for a given "Card".
+func CountMatches(cardData string) int {
+	winnersAndReveals := strings.Split(strings.Split(cardData, ": ")[1], " | ")
+
+	revealed := make(map[string]bool)
+
+	for _, v := range strings.Split(winnersAndReveals[1], " ") {
+		if v != "" {
+			revealed[v] = true
+		}
+	}
+
+	matches := 0
+
+	for _, v := range strings.Split(winnersAndReveals[0], " ") {
+		if v == "" {
+			continue
+		}
+		if revealed[v] {
+			matches++
+		}
+	}
+	return matches
+}
+
+// SumAllPoints returns the total score using all of the matches found for every "Card"
+// in the text file at the provided path location.
+func SumAllPoints(path string) (int, error) {
+	file, err := os.Open(path)
+	if err != nil {
+		return 0, err
+	}
+	defer file.Close()
+
+	score := 0
+	scanner := bufio.NewScanner(file)
```

Comment:
I've been doing a bad thing in previous problems, loading the entire dataset into memory... But for this problem, we're streaming in the data one line at a time, minimizing our memory usage. We could definitely improve the performance here with some concurrency, but I'll leave that for later.